packages/sdk-cli/src/models/HostConnections.test.ts

```diff
@@ -23,22 +24,24 @@ function mockSentinel(spy: jest.SpyInstance) {
 beforeEach(() => {
   hostConnections = new HostConnections();
   hostConnections.onHostAdded.attach(hostConnectedSpy);
-  relayConnectSpy = jest.spyOn(developerRelay, 'connect');
+  relayConnectSpy = jest.spyOn(DeveloperRelay.prototype, 'connect');
   remoteHostSpy = jest.spyOn(RemoteHost, 'connect');
 });
 
-function doConnect(type: HostType) {
-  return hostConnections.connect(type, mockHostID);
+async function doConnect(type: HostType) {
+  jest.spyOn(relayInstance, 'connect').mockResolvedValueOnce(new Duplex());
+  const ws = await relayInstance.connect(mockHostID);
+  return hostConnections.connect(type, ws);
 }
 
 describe.each<HostType>(['appHost', 'companionHost'])(
   'when the host type argument is %s',
   (hostType) => {
-    let mockWS: jest.Mocked<stream.Duplex>;
+    let mockWS: jest.Mocked<Duplex>;
     let mockRemoteHost: {};
 
     beforeEach(() => {
-      mockWS = mockSentinel(relayConnectSpy) as jest.Mocked<stream.Duplex>;
+      mockWS = mockSentinel(relayConnectSpy) as jest.Mocked<Duplex>;
       mockWS.destroy = jest.fn();
 
       mockRemoteHost = mockSentinel(remoteHostSpy);
```

Comment:
I can't comment on the specific line because it hasn't changed, but the `acquires a developer relay connection for the given host ID` test doesn't really make sense anymore: it's testing test code. There's probably a new place that test belongs based on the refactoring you've done.